Task5/Task5_2/Catalog.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using static System.Reflection.Metadata.BlobBuilder;
+
+namespace Task5_2
+{
+    internal class Catalog<isbn, book>
+    {
+        private Dictionary<isbn, book> _catalog;
+
+        public book this[isbn key]
+        {
+            get
+            {
+                key = simplifyString(key);
+
+                if (_catalog.ContainsKey(key))
+                {
+                    return _catalog[key];
+                }
+                else
```

Comment:
For better performance just return null, not exception

Reply:
Modified